airbyte-integrations/connectors/source-snapchat-marketing/source_snapchat_marketing/source.py

```diff
@@ -228,13 +223,9 @@ def get_updated_state(self, current_stream_state: MutableMapping[str, Any], late
         Thus all the slices data are compared to the initial state, but only on the last one we write it to the stream state.
         This approach gives us the maximum state value of all the records and we exclude the state updates between slice processing
         """
-
-        if self.first_run:
-            self.first_run = False
-            return {self.cursor_field: self.initial_state}
-        else:
-            self.max_state = max(self.max_state, latest_record[self.cursor_field])
-            return {self.cursor_field: self.max_state if self.current_slice == self.last_slice else self.initial_state}
+        if not current_stream_state:
+            current_stream_state = {self.cursor_field: self.start_date}
+        return {self.cursor_field: max(latest_record.get(self.cursor_field, ""), current_stream_state.get(self.cursor_field, ""))}
```

Comment:
yes. In case, 
`current_stream_state: {'updated_at': '2021-06-11T09:44:42.583Z'}` and 
`latest_record.get(self.cursor_field, "") = ""` max return '2021-06-11T09:44:42.583Z'